corehq/util/django_management.py

```diff
@@ -0,0 +1,34 @@
+from django.core.management import BaseCommand
+from django.core.management.base import SystemCheckError
+
+from corehq.util.datadog.utils import create_datadog_event
+
+
+class AuditedBaseCommand(BaseCommand):
+    command_name = None
+
+    def create_parser(self, prog_name, subcommand):
+        self.command_name = prog_name
+        return super(AuditedBaseCommand, self).create_parser(prog_name, subcommand)
+
+    def execute(self, *args, **options):
+        self.create_datadog_event('start', args, options)
+        try:
+            output = super(AuditedBaseCommand, self).execute(*args, **options)
+        except SystemCheckError:
+            raise
+        except Exception as e:
+            self.create_datadog_event('stop', args, options, e)
+            raise
+        else:
+            self.create_datadog_event('stop', args, options)
+            return output
+
+    def create_datadog_event(self, start_stop, args, options, error=None):
+        text = 'args: {}\noptions: {}'.format(args, options)
```

Comment:
Is there any danger of sending sensitive information to datadog with this?

Reply:
I haven't done an audit but I don't think so. If passwords etc are required they are collected via stdin and not as args.